src/ott/geometry/regularizers.py

```diff
@@ -0,0 +1,531 @@
+
+  Args:
+    f: Function :math:`f` applied to :math:`Ax`.
+    A: Linear operator :math:`A`.
+    b: Offset :math:`b`. If :obj:`None`, use array of 0s.
+    nu: Value for which :math:`AA^T = \nu I` holds.
+  """
+
+  def __init__(
+      self,
+      f: ProximalOperator,
+      A: Optional[Union[jnp.ndarray, lx.AbstractLinearOperator]],
+      b: Optional[jnp.ndarray] = None,
+      nu: float = 1.0,
+  ):
+    assert nu > 0.0, nu
+    super().__init__()
+    self.f = f
+    # AA^T = alpha I
+    self.A = lx.MatrixLinearOperator(A) if isinstance(A, jnp.ndarray) else A
+    self.b = b
+    self.nu = nu
+
+  def __call__(self, x: jnp.ndarray) -> float:  # noqa: D102
+    z = self.A.mv(x)
+    if self.b is not None:
+      z = z + self.b
+    return self.f(z)
+
+  def prox(self, v: jnp.ndarray, tau: float = 1.0) -> jnp.ndarray:  # noqa: D102
+    w = self.A.mv(v)
+    if self.b is None:
+      tmp = self.f.prox(w, tau * self.nu)
+    else:
+      tmp = self.f.prox(w + self.b, tau * self.nu) - self.b
+    return v - (1.0 / self.nu) * (self.A.T.mv(w - tmp))
+
+  @property
+  def is_fully_orthogonal(self) -> bool:
+    r"""Whether :math:`\nu = 1`."""
+    return self.nu == 1.0
+
+  def tree_flatten(self):  # noqa: D102
+    return (self.f, self.A, self.b), {"nu": self.nu}
+
+
+@jtu.register_pytree_node_class
+class Quadratic(ProximalOperator):
+  r"""Quadratic operator :math:`\frac{1}{2} \left<x, Q x\right> + b`.
+
+  .. note::
+    This constructor is not meant to be called by the user,
+    please use the :meth:`create` method instead.
+
+  Args:
+    A: Linear operator :math:`A`. If :obj:`None`, use identity.
+    A_comp: Orthogonal complement :math:`A^{\perp}` of :math:`A`, if computed.
+    b: Offset :math:`b`. If :obj:`None`, use array of 0s.
+    solver: Linear solver.
+  """
+
+  def __init__(
+      self,
+      A: Optional[lx.TaggedLinearOperator],
+      A_comp: Optional[lx.AbstractLinearOperator] = None,
+      b: Optional[jnp.ndarray] = None,
+      solver: Optional[Callable[[lx.AbstractLinearOperator, jnp.ndarray],
+                                jnp.ndarray]] = None,
+  ):
+    super().__init__()
+    self.A = A
+    self.A_comp = A_comp
+    self.b = b
+    self.solver = (
+        lambda op, b: lx.linear_solve(op, b).value
+    ) if solver is None else solver
+
+  @classmethod
+  def create(
+      cls,
+      A: Optional[Union[jnp.ndarray, lx.AbstractLinearOperator]] = None,
+      *,
+      is_complement: bool = False,
+      is_orthogonal: bool = False,
+      is_squared: bool = False,
+      **kwargs: Any,
+  ) -> "Quadratic":
+    r"""Create the quadratic operator :math:`\frac{1}{2} \left<x, Q x\right> + b`.
+
+    The :math:`Q` is defined as:
+
+    - :math:`Q := A` if not squared and not a complement.
+    - :math:`Q := A^{\perp}` if not squared and a complement.
+    - :math:`Q := A^TA` if squared and not a complement.
+    - :math:`Q := \left(A^{\perp}\right)^TA^{\perp}` if squared and
+      a complement.
```

Comment:
What's Q if A is orthogonal? the last one?

Reply:
No, it's partially explained in the `is_orthogonal` section of the docs. Will mention that in such case, no linear solve/matrix inversion is necessary if `is_squared=True`.